File: c++ files/reversearray.cpp
#include <iostream>
using namespace std;

void reverseArray(int arr[], int size){
    int start=0, end= size-1;
    while (start<=end){
        swap(arr[start], arr[end]);
        start++;
        end--;
    }
    
}

void printArray(int arr[], int size){
    for (int i=0; i<size; i++){
        cout << arr[i] <<" ";
    }
}


int main(){
    int s;
    int arr[100];
    
    cin >> s;

    if (s==1){
        cout << "Enter more than one value" <<endl;
    }

    else if (s == 0){
        cout << "Wrong Input" <<endl;
    }

    else if (s>1){

        for (int i=0; i<s; i++){
        cin >> arr[i];
        }

        reverseArray(arr, s);
        printArray(arr, s);
    }
    

    return 0;
}

File: java files/reversearray.java
public class reversearray {

    public static void swap(int a, int b){
        int temp=0;
        temp=a;
        a=b;
        b=temp;
        System.out.print(b + a);
    }
    
    public static void reverse(int arr[], int n){
        int start=0;
        int end=n-1;

        while(start<=end){
            swap(arr[start], arr[end]);
            start++;
            end--;
        }
    }

    public static void getArray(int[] arr){
        int n=arr.length;
        for(int i=0; i<n; i++){
            System.out.println(arr[i]);
        }
    }
    
    public static void main(String[] args) {
        int[] arr = {2, 6, 7, 8, 5};

        System.out.println("The Given Array is : ");
        getArray(arr);

        reverse(arr, 5);
        System.out.println("The Reversed Array is :");
        getArray(arr);
        
    }
}
